pkg/identity/identity.go

```diff
@@ -181,6 +181,10 @@ func PeerIdentityFromCerts(leaf, ca *x509.Certificate, rest []*x509.Certificate)
 
 // PeerIdentityFromPeer loads a PeerIdentity from a peer connection
 func PeerIdentityFromPeer(peer *peer.Peer) (*PeerIdentity, error) {
+	if peer.AuthInfo == nil {
+		return nil, Error.New("peer AuthInfo is nil")
+	}
+
 	tlsInfo := peer.AuthInfo.(credentials.TLSInfo)
```

Comment:
Let's also check for this conversion.

Reply:
Do we actually need both, I forgot about checking the second return arg - wouldn't that be sufficient on it's own?

Reply:
The problem with not checking the conversion is that it will panic otherwise. So if a user can figure out how to get a different auth into the system, then it would panic taking down the whole server. (e.g. some grpc change upstream)